### PR TITLE
updates to case OData format

### PR DESCRIPTION
**Pull request by nickpell**
🐡 

This PR addresses [two issues](https://trello.com/c/ujKXg4mz/73-commcare-data-3rd-party-reporting-tool):
1) For expected case properties missing from a case, represents this with null, so PowerBI doesn't claim that there's an error with the row.
2) Avoids nesting in the row, which means we can add back underscores in case properties (Tableau doesn't like underscores in nested structures)

After these changes are deployed we will need to tell people in the beta group to refresh their metadata documents.

### CODE REVIEW

File: corehq/apps/api/odata/tests/data/populated_case_metadata_document.xml [hunk 1]
```diff
@@ -1,22 +1,6 @@
...
-         <Property Name="owner_id" Type="Edm.String"/>
-       </ComplexType>
-       <ComplexType Name="CaseProperties:case_type_with_case_properties">
-         <Property Name="backend_id" Type="Edm.String"/>
-         <Property Name="case_name" Type="Edm.String"/>
-         <Property Name="case_type" Type="Edm.String"/>
-         <Property Name="date_opened" Type="Edm.String"/>
-         <Property Name="owner_id" Type="Edm.String"/>
-         <Property Name="property_1" Type="Edm.String"/>
-         <Property Name="property_2" Type="Edm.String"/>
```
Comment (czue) [inline corehq/apps/api/odata/tests/data/populated_case_metadata_document.xml]: I'm not sure it matters, and I think I'm just misunderstanding something, but why were there underscores in these to begin with? Should they not have previously been replaced with empty strings?

**Review by czue — APPROVED** [review]
Looks good!

> After these changes are deployed we will need to tell people in the beta group to refresh their metadata documents.

Is this something that needs to be coordinated? @djmore9

File: corehq/apps/api/odata/serializers.py [hunk 2]
```diff
@@ -62,10 +62,11 @@ def update_case_json(self, case_json, domain, case_type):
             'resource_uri',
         ]:
             case_json.pop(remove_property)
-        case_json['properties'] = {
-            property_name: case_json['properties'].get(property_name, None)
+        case_properties = case_json.pop('properties')
+        case_json.update({
+            property_name: case_properties.get(property_name, None)
```
[the remark this answers is not in the document]
Comment (nickpell) [reply corehq/apps/api/odata/serializers.py]: I don't think so, for the reason you mentioned.

File: corehq/apps/api/odata/tests/data/populated_case_metadata_document.xml (hunk 1 above)
Comment (nickpell) [reply corehq/apps/api/odata/tests/data/populated_case_metadata_document.xml]: The test mocked out the function that remove the empty strings, which is why they got passed through to the metadata document.

**Pull request merged**